### DISCUSSION TITLE
[Q+A] Running in Windows subsystem for Linux

### DISCUSSION THREAD

**Discussion by espr14** [vintagepc/MK404#296] [Q&A]
I can build MK404 on Ubuntu 20 on WSL on ext4. WSL (compared to WSL2) does not have display:0 but I used VNC so I ran it on display:1. It has some problem with OpenGL. I'll try WSL2.

**Comment by vintagepc** [comment]
What sort of issue are you encountering? (error messages, console output etc)? 

Have you tried with the "native" cygwin build?

**Comment by 3d-gussner** [comment]
@espr14 I had MK404 running in WSL2 and it kind of worked. Changed to Dual boot due to my "old" PC wasn't performing that well with MK404 in WSL2.
Also to install MK404 on "fresh" Ubuntu this might help:
```
sudo apt install -y libelf-dev gcc-7 gcc-avr libglew-dev freeglut3-dev libsdl-sound1.2-dev libpng-dev cmake build-essential lcov mtools
```

**Comment by espr14** [comment]
sudo apt install git
and finally running on VirtualBox Ubuntu 20.04.

**Comment by vintagepc** [comment]
That should be fine for most purposes, though the VM likely will not have enough GPU grunt for `-g fancy` or any of the 3d extrusion options.

**Comment by espr14** [comment]
`-g lite` runs ~8 fps.

**Comment by vintagepc** [comment]
Have you enabled 2d/3d acceleration in Virtualbox and installed the drivers in the ubuntu guest?

**Comment by vintagepc** [comment]
https://blogs.oracle.com/scoter/oracle-vm-virtualbox-6-3d-acceleration-for-ubuntu-1804-guest
